src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,114 @@
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d("ZCurrentVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kBlue)));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d("TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
+
+    public SpeedMechanism2d(String key, double maxDisplayableVelocity, double deadband) {
+        this.deadband = deadband;
+        this.key = key;
+        this.mechanism = new Mechanism2d(2 * maxDisplayableVelocity, 2 * maxDisplayableVelocity);
+        MechanismRoot2d root = mechanism.getRoot("VelocityRoot", maxDisplayableVelocity, maxDisplayableVelocity);
+        this.currentVelocityLigament = root.append(new MechanismLigament2d("ZCurrentVelocityLigament", 0, 0, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityTopArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(  "ZCurrentVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d("ZCurrentVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kBlue)));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d("TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
+
+    public void setVelocity(double velocity) {
+        setVelocity(velocity, lastTargetVelocity);
+    }
+
+    public void setVelocity(double velocity, double targetVelocity) {
+        setTargetVelocity(targetVelocity);
+        setArrowAngle(velocity, targetVelocity);
+        currentVelocityLigament.setLength(velocity);
+        setCurrentLigamentColor(velocityToColor(velocity));
+        Logger.recordOutput(key, mechanism);
+    }
```

Comment:
1. Call these methods `update` to explain that they also log the mechanism. 
2. The `setVelocity(double velocity, double targetVelocity)` should call `setVelocity(double velocity)` (to be renamed to `update`) and `setTargetVelocity(double targetVelocity)`. And the `setVelocity(double velocity)` (to be renamed to `update`) should handle setting the current ligaments angles vel etc. 
3. No need for the `lastTargetVelocity` since it's "stored" already inside the target ligament
4. Add javadocs to both methods